crates/services/p2p/src/peer_report.rs

```diff
@@ -305,77 +273,108 @@ impl NetworkBehaviour for PeerReportBehaviour {
         &mut self,
         peer_id: PeerId,
         connection_id: ConnectionId,
-        event: <<Self::ConnectionHandler as IntoConnectionHandler>::Handler as
-            ConnectionHandler>::OutEvent,
+        event: THandlerOutEvent<Self>,
     ) {
         match event {
-            EitherOutput::First(heartbeat_event) => self
-                .heartbeat
-                .on_connection_handler_event(peer_id, connection_id, heartbeat_event),
-            EitherOutput::Second(identify_event) => self
-                .identify
-                .on_connection_handler_event(peer_id, connection_id, identify_event),
+            Either::Left(heartbeat_event) => self.heartbeat.on_connection_handler_event(
+                peer_id,
+                connection_id,
+                heartbeat_event,
+            ),
+            Either::Right(identify_event) => self.identify.on_connection_handler_event(
+                peer_id,
+                connection_id,
+                identify_event,
+            ),
         }
     }
+
+    fn handle_established_inbound_connection(
+        &mut self,
+        _connection_id: ConnectionId,
+        peer: PeerId,
+        local_addr: &Multiaddr,
+        remote_addr: &Multiaddr,
+    ) -> Result<THandler<Self>, ConnectionDenied> {
+        self.identify
```

Comment:
This is a guess.

Another approach would be to do some kind of check with `identify` to see if it's already handled... and then pass it to `heartbeat` in that case. I'm not sure how important this is though.